src/FulfillerMultiSelect.jsx

```diff
@@ -119,20 +148,53 @@ class FulfillerMultiSelect extends React.Component {
     async getRecentFulfillerIds() {
         if (this.props.accessToken) {
             let settings = await this.customizrClient.getSettings(this.props.accessToken);
-            let recentFulfillerIds = settings.recentFulfillerIds || [];
-            this.setState({ recentFulfillerIds });
-            return recentFulfillerIds;
+            return {
+                recentFulfillerIds: settings.recentFulfillerIds || [],
+                recentFulfillerIdsMulti: settings.recentFulfillerIdsMulti || []
+            };
+        }
+    }
+
+    recentFulfillersChanged(fulfillerIds) {
+        const recentFulfillerIds = this.state.recentFulfillerIds;
+
+        if (recentFulfillerIds.length === 0) {
+            return false;
+        }
+
+        if (fulfillerIds > recentFulfillerIds) {
+            return true;
+        }
+
+        if (this.props.multi) {
+            return fulfillerIds.length !== recentFulfillerIds.length
+                || fulfillerIds.any((fulfillerId, i) => recentFulfillerIds[i] === fulfillerId);
         }
+
+        return  fulfillerIds[0] !== recentFulfillerIds[0];
     }
 
     async updateRecentFulfillerIds(fulfillerIds) {
+        if (!this.recentFulfillersChanged(fulfillerIds)) {
+            return;
+        }
+
         // initial, strictly visual client-side update to circumvent a wait on GET Customizr
         this.setState({
-            recentFulfillerIds: fulfillerIds
+            recentFulfillerIds: fulfillerIds.concat((this.state.recentFulfillerIds || []).filter(id => !fulfillerIds.includes(id)))
         });
 
-        // consistent server-side update
-        let update = { recentFulfillerIds: fulfillerIds };
+        let { recentFulfillerIds, recentFulfillerIdsMulti } = (await this.getRecentFulfillerIds()) || [];
```

Comment:
Somewhere around this area of code, you could reverse the recent fulfiller list. This way, the Fulfiller(Single)Select would correctly display the most recent fulfiller.

You would also have to reverse the list as you load it into the FulfillerMultiSelect.

Reply:
Done, thanks!